### ISSUE TITLE
Invalid escape \'

### ISSUE THREAD

**Issue by cburgmer** [gocd/gocd#1472]
A file name containing a `'` is incorrectly escaped as `\'` in the artefact tree JSON response for the API `/go/files/PIPELINE/12/STAGE/34/JOB.json`.

See example https://github.com/cburgmer/buildviz/blob/master/test/gosync/__files/job_files_with_encoding_error.json.

Python's JSON parse throws:
```
$ cat job_files_with_encoding_error.json | python -m json.tool
Invalid \escape: line 1 column 44 (char 43)
```

**Comment by ketan** [comment]
Replicating the JSON payload here for future reference in case @cburgmer kills the repo -

```json
[ { "name": "Random File with invalid escap\'e","url": "http://localhost:1234/files/A_Pipeline/41/A_Stage/1/A_Job/a_directory/random_file.log" } ]
```

**Comment by magnus-lycka** [comment]
I get the same problem when I try to post to "/go/admin/restful/configuration/file/POST/xml" .
If the post contains e.g. two pipelines with the same name, the "result" in the response will be:

You have defined multiple pipelines called 'pipeline_name'. Pipeline names must be unique.

This get encoded as "... \'pipeline_name\' ..." which is in violation of the JSON standard RFC 7159.

**Comment by rajiesh** [comment]
verified this fix on 15.3.0-2690(4fd0926ef27e6e19428b62ffeb2931d834069c01) and the json with quotes are escaped with %27 encoded char. also deployed this build on build.go.cd and verified the pipeline history page is rendered properly
